Requirement 1.2.5 Product Breakdown

aimeekang opened the pull request:
Requirement 1.2.5 Product Breakdown

aimeekang committed:
save styling to switch to review branch

aimeekang committed:
Merge branch 'main' into RatingsReviews

aimeekang committed:
styling (mainly review list / review tile)

aimeekang committed:
product breakdown styling

aimeekang committed:
implement star ratings via react icons

aimeekang committed:
Merge branch 'main' into RatingsReviews

AedonGrunGott commented on client/source/components/RR/ReviewTile.jsx:
```diff
@@ -60,66 +52,71 @@ const ReviewTile = function ReviewTile({ review, updateList }) {
-              onClick={() => setFullBody(true)}
-            >SHOW MORE
+              onClick={handleHelpfulness}
+            >Yes {`(${review.helpfulness})`}
```
Not a merge blocker. Just by switching between sort options, I am able to set something as helpful multiple times.

aimeekang replied:
Hm, that is a good point - I have sortType set as a state, so every time it changes, it re-renders the list, as I'm using the sort parameter of the API. I'll probably have to look in how to cache the data. Thoughts?

AedonGrunGott commented on client/source/components/RR/ProductBreakdown.jsx:
```diff
@@ -8,26 +9,28 @@ const ProductBreakdown = function ProductBreakdown({ traits }) {
+          ? (<div><Slider trait="Fit" feedback={traits.Fit.value} key={traits.Fit.id} /></div>)
+          : null }
+      </div>
+    </ProductBreakdownContainer>
```
Is Fresh & Clean! 🧼

AedonGrunGott commented on client/source/components/RR/ReviewRating.jsx:
```diff
@@ -0,0 +1,27 @@
+  );
+};
+
+export default ReviewRating;
```
Placeholder, to be replaced! Works as is though, so useful placeholder.

AedonGrunGott commented on client/source/components/RR/ReviewTile.jsx:
```diff
@@ -60,66 +52,71 @@ const ReviewTile = function ReviewTile({ review, updateList }) {
-      <div className="review-body-container">
-        <div className="review-text">{capBody(review.body)}</div>
-        { review.body.length > 250 && fullBody === false
+    <ReviewTileContainer>
```
Not a merge blocker. There is a _lot_ going on in this render section. I'd suggest breaking it up into smaller sub-components.

AedonGrunGott reviewed (approved):
![Approved-stamp-illustration-on-transparent-background-PNG](https://user-images.githubusercontent.com/16231955/216432926-47d15558-ff84-4657-acd1-6d5de04e3645.png)

aimeekang committed:
restructure data

aimeekang committed:
refactor to have meta data live in AppHolder, and create helpers.js file for shared code

aimeekang commented:
@BrettEastman or @avinashi10 Would you mind taking a look at the latest commits after I got August's approval? 

Main changes: I restructured the meta data that is needed to calculate average rating, added a file for helper functions (currently only contains a function to calculate the average rating), and updated the code on the Review Module so it doesn't repeat with what's happening in the App Holder. The plan is that the meta data is exposed via the context provider, and whoever needs to calculate the average rating can use the helpers.js file for it.

BrettEastman commented:
I checked it out and it looks good to me. I tested the helpful button and it only incremented once, which seems good.
